src/native/s3_client.c

```diff
@@ -489,7 +490,8 @@ static void s_on_s3_meta_request_finish_callback(
             meta_request_result->response_status,
             jni_payload,
             meta_request_result->validation_algorithm,
-            meta_request_result->did_validate);
+            meta_request_result->did_validate,
+            callback_data->java_exception);
```

Comment:
Maybe we should only use the exception from callback on a certain error code in C.

eg: current impl will raise error like:
```
CRT_ERROR: socket timeout
caused by: File Open Failure from java
```

Reply:
java_exception will only be set if we receive any exception from Java in the onHeaders or onBody callback. I am not sure how socket timeout will have a cause.

Reply:
We raise HTTP_CALLBACK_FAILURE in the two places where we are setting the java_exception so other errors will not have a cause.

Reply:
It's possible that before the `HTTP_CALLBACK_FAILURE` got propagated to the meta request failure, something else failed from any other thread, and override the callback failure

Reply:
Thanks, updated.